reVX/setbacks/setbacks_cli.py

```diff
@@ -2,392 +2,552 @@
+        Collection config. This config will be updated to include the
+        keys "_out_path" and "_pattern" representing the output file
+        path and the input file pattern, respectively.
+    project_dir : path-like
+        Path to project directory. This path is used to resolve the
+        output filepath.
+    command_name : str
+        Name of the command being run. This is used to parse the
+        pipeline status for output files if
+        ``"merge_file_pattern": "PIPELINE"`` in the input `config`.
+    merge_file_pattern : str | list | dict, optional
+        Unix-style ``/filepath/pattern*.h5`` representing the files to
+        be merged into a single output GeoTIFF file. If no output file
+        path is specified (i.e. this input is a single pattern or a list
+        of patterns), the output file path will be inferred from the
+        pattern itself (specifically, the wildcard will be removed
+        and the result will be the output file path). If a list of
+        patterns is provided, each pattern will be merged into a
+        separate output file. To specify the name of the output file(s),
+        set this input to a dictionary whose keys are paths to the
+        output file (relative paths are allowed) and the values are
+        patterns representing the input files that should be merged into
+        the output TIFF. If running a merge job as part of a pipeline,
+        this input can be set to ``"PIPELINE"``, which will parse the
+        output of the previous step (``compute``) and generate the input
+        file pattern and output file name automatically.
+        By default, ``"PIPELINE"``.
+
+    Returns
+    -------
+    config : dict
+        Updated merge config dictionary.
     """
-    if feature_type not in SETBACKS:
-        msg = ("feature_type must be one of: {}; got {!r}"
-               .format(set(SETBACKS.keys()), feature_type))
-        raise RuntimeError(msg)
+    return preprocess_collect_config(config, project_dir, command_name,
+                                     collect_pattern=merge_file_pattern)
 
-    if 'VERBOSE' in ctx.obj:
-        verbose = any((ctx.obj['VERBOSE'], verbose))
 
-    log_modules = [__name__, 'reVX', 'reV', 'rex']
-    init_mult(ctx.obj['NAME'], log_dir, modules=log_modules, verbose=verbose)
+def _update_setbacks_calculators(feature_specs=None):
+    """Update `SETBACKS` to include new specs """
 
-    # same check as the config in case someone invokes this from the
-    # direct command line instead of a config file for some bizarre reason
-    validate_setback_regulations_input(base_setback_dist, hub_height,
-                                       rotor_diameter)
+    for feature_name, feature_spec in (feature_specs or {}).items():
+        spec = SETBACK_SPECS.get(feature_name, {})
+        spec.update(feature_spec)
+        SETBACKS[feature_name] = setbacks_calculator(**spec)
+
+
+def compute_setbacks(excl_fpath, _ft, _fp, _mult, out_dir, tag,
```

Comment:
I typically use a leading underscore to "hide" attributes of a class, what's the reason for underscore inputs like _ft and _fp? If this is a GAPS thing, any way we can actually hide these inputs as kwargs or something? Also, docstring does not match the actual variable names.

Reply:
Yea essentially these are arguments that need to be explicit (they need to be detectable with `inspect.signature` - `**kwargs` would unfortunately obscure them) but shouldn't be exposed to the end-user documentation since they will always be provided by the code based on other user inputs.

I'm on the fence about this syntax. I think it's explicit and indicates the intent at a glance (as long as you are aware this is how GAPs works), but also it's kind of ugly and TBH probably doesn't belong in a public function. I think based on this feedback, I will explore ways to "hide" parameter names in the command setup class rather than relying on syntax. I think this idea should work - if not I will ping you to brainstorm other solutions

Reply:
We should definitely touch base as to why these "hidden" parameters are required. In the meantime, do you think he updated syntax suffices?